### PR TITLE
Development & Release configuration + xcode schema run from xcode , display name, bundle identifier, run from vs code fix, improve read me etc.

### PR DESCRIPTION
**Pull request by dinkar1708**
**What**  
- Added simple pr template
- Fixed to run from vs code
- Fixed to run from xcode
- Fixed app bundler identifier and display name
- Keep only Development and Release mode same as android
  Development -> com.monstarlab.oss.flutter.template.flutter-template-riverpod-debug
  Release -> com.monstarlab.oss.flutter.template.flutter-template-riverpod

**Screen shots**  
Xcode config
Screen Shot 2022-11-08 at 15 09 47
Xcode config
Screen Shot 2022-11-08 at 15 09 33

Printed logs to show which mode is running
image

Launcher app name
Simulator Screen Shot - iPhone Xs Max - 2022-11-08 at 15 22 15

Run config file 
image
**Api logs**  
- NA

**Other logs**  
Why Debug-Development and Release-Production names
when run using vs code shows below 
The Xcode project defines build configurations: Debug, Release
Flutter expects a build configuration named Debug-Development or similar.
Open Xcode to fix the problem:
  open ios/Runner.xcworkspace

### CODE REVIEW

**Comment by dinkar1708** [comment]
Highlighted is the main fix to run files...
image

File: .github/pull-request-template.md [hunk 1]
```diff
@@ -0,0 +1,9 @@
+
+**What**  
+- 
+
+**Screen shots**  
+- NA
+
+**Api logs** 
+- NA
```
Comment (enricmacias) [inline .github/pull-request-template.md]: Maybe we can discuss this in another PR? What kind of PR template we need for the team

File: README.md [hunk 2]
```diff
@@ -1,37 +1,67 @@
...
+### run iOS/Android From terminal
+```
+- flutter run --flavor Development -t lib/main/main_development.dart
+```
 
-Launch Configurations are included in the `.run` folder. 
\ No newline at end of file
+### run iOS using xcode
+- Select Development schema from top center and run in development mode(Debug)
+- Should select Production for archive release mode ipa file
```
Comment (enricmacias) [inline README.md:48]: The README changes are completely unrelated to the environments.  Can we better handle this in another PR?

File: README.md [hunk 3]
```diff
@@ -1,37 +1,67 @@
...
 flutter pub get
 flutter packages pub run build_runner build --delete-conflicting-outputs
 ```
 
-### Visual Studio
-
-Launch Configurations are included in the `.vscode/launch.json` file.
+## Run 
+### Run iOS/Android using Visual Studio
+- create launch.json and add add below
```
Comment (enricmacias) [inline README.md]:
```suggestion
- create launch.json and add below
```

File: ios/Runner.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -361,22 +361,23 @@
 			};
 			name = "Debug-Development";
 		};
-		82B3057D290109C5000CC069 /* Debug-Development */ = {
+		82B3057F290109E5000CC069 /* Debug-Development */ = {
 			isa = XCBuildConfiguration;
 			baseConfigurationReference = 9740EEB21CF90195004384FC /* Debug.xcconfig */;
 			buildSettings = {
-				APP_DISPLAY_NAME = "[DEV] Flutter Template Riverpod";
+				APP_DISPLAY_NAME = "DevFlutter Template Riverpod";
```
Comment (enricmacias) [inline ios/Runner.xcodeproj/project.pbxproj]: Did you change this name? I think [DEV] Flutter Template Riverpod is better than DevFlutter

File: ios/Runner.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -595,12 +442,13 @@
 			isa = XCBuildConfiguration;
 			baseConfigurationReference = 7AFA3C8E1D35360C0083082E /* Release.xcconfig */;
 			buildSettings = {
-				APP_DISPLAY_NAME = "Flutter Template Riverpod";
+				APP_DISPLAY_NAME = "ProdFlutter Template Riverpod";
```
Comment (enricmacias) [inline ios/Runner.xcodeproj/project.pbxproj]: `[PROD] Flutter Template Riverpod` should be better ?

**Review by enricmacias — CHANGES_REQUESTED** [review]
Left some comments

File: .github/pull-request-template.md (hunk 1 above)
Comment (dinkar1708) [reply .github/pull-request-template.md]: in same pr we can discuss there is no hurry
i thought to keep simple template by which we can understand whats going on in pr at least 
easy for review.

would you like to change anything from above?

File: README.md (hunk 2 above)
Comment (dinkar1708) [reply README.md:48]: i think we can do in same pr because i wrote about what changes i did.
also if you think any suggestion i can update read me.

File: ios/Runner.xcodeproj/project.pbxproj (hunk 4 above)
Comment (dinkar1708) [reply ios/Runner.xcodeproj/project.pbxproj]: yes i changed with [] now for both dev and prod

File: .github/pull-request-template.md (hunk 1 above)
Comment (diarmaidlindsay) [reply .github/pull-request-template.md:9]: Ok since we are on the topic, in my opinion this one is much more detailed and includes guidance for the submitter, but it will depend on the project.

```
### Issue

- close `(required: Put the ticket link)`

### Motivation and Context

- `(required)`

### Modified points

- [ ] `(required: Write "What you did" here)`

### Description and Screenshots

- `(optional)`

### Figma or screen image URL

- `(required)`

### Worries or Issues

- `(optional)`
```

File: .vscode/launch.json [hunk 6]
```diff
@@ -5,24 +5,25 @@
   "version": "0.2.0",
   "configurations": [
     {
-      "name": "flutter_template_riverpod",
+      "name": "flutter_template_riverpod(development mode)",
```
Comment (diarmaidlindsay) [inline .vscode/launch.json]: Include a space as below please.
`flutter_template_riverpod (development mode)`
Same as you did with "release mode".

File: .github/pull-request-template.md (hunk 1 above)
Comment (diarmaidlindsay) [reply .github/pull-request-template.md]: Ok since we are on the topic, in my opinion this one is much more detailed and includes guidance for the submitter, but it will depend on the project.

```
### Issue

- close `(required: Put the ticket link)`

### Motivation and Context

- `(required)`

### Modified points

- [ ] `(required: Write "What you did" here)`

### Description and Screenshots

- `(optional)`

### Figma or screen image URL

- `(required)`

### Worries or Issues

- `(optional)`
```

File: .vscode/launch.json [hunk 7]
```diff
@@ -5,24 +5,25 @@
...
     {
       "name": "flutter_template_riverpod (release mode)",
       "request": "launch",
       "type": "dart",
       "flutterMode": "release",
-      "program": "lib/main/main_production.dart"
+      "program": "lib/main/main_production.dart",
+      "args": [
+        "--flavor",
+        "Release",
```
Comment (diarmaidlindsay) [inline .vscode/launch.json]: The 2 Flavors are "Development" and "Production".
You are confusing between "flavors" and "build modes" (`flutterMode`) by calling this "release".
So please rename this flavor to "Production".

Please refer to the docs : https://docs.flutter.dev/testing/build-modes

So in total there will be 6 app versions...
Development-Debug
Development-Profile
Development-Release
Production-Debug
Production-Profile
Production-Release

However I acknowledge that this `launch.json` currently only has 2 app versions listed.
I think it would be nice to add a "Development-Debug" launch option for VSCode users actually, because this is most common.

File: README.md (hunk 2 above)
Comment (diarmaidlindsay) [reply README.md:48]: You threw away all of the comments in my old Readme file which explained the basis for creating this template, and replaced it with the default Readme template?! You also removed the Android Studio section.
I will not approve these changes to the Readme unless you restore those parts.

File: lib/main/main_common.dart [hunk 8]
```diff
@@ -8,6 +8,7 @@ import '../presentation/app.dart';
 import 'app_flavor.dart';
 
 void mainCommon(AppFlavor appFlavor) {
+  debugPrint('launching....mode.... ${appFlavor.appFlavorType}');
```
Comment (diarmaidlindsay) [inline lib/main/main_common.dart]: I would like to present proper capitalisation and punctuation to the rest of MonstarLab members who will use this. Please change to : `Launching Mode : ${appFlavor.appFlavorType}`.

**Review by diarmaidlindsay — CHANGES_REQUESTED** [review]

File: .vscode/launch.json (hunk 7 above)
Comment (dinkar1708) [reply .vscode/launch.json]: @diarmaidlindsay i am keeping only two not including profile just to make same as android.

File: README.md (hunk 2 above)
Comment (dinkar1708) [reply README.md:48]: hey sorry it was by mistakes seems some temporary mistakes i fixed.

File: .vscode/launch.json (hunk 7 above)
Comment (dinkar1708) [reply .vscode/launch.json]: build mode i kept only two debug and release
you want me to include profile mode? i ignored profile to keep same as android
https://docs.flutter.dev/testing/build-modes

Comment (dinkar1708) [reply .vscode/launch.json]: >So please rename this flavor to "Production".

changed.

File: README.md [hunk 9]
```diff
@@ -19,19 +19,61 @@ it ideal for a demo/template.
...
-### Visual Studio
+## Run 
+### Android Studio
+Launch Configurations are included in the `.run` folder. 
 
-Launch Configurations are included in the `.vscode/launch.json` file.
 
-### Android Studio
+### Run iOS/Android using Visual Studio
+- create launch.json and add below
```
Comment (diarmaidlindsay) [inline README.md]: Since we will commit the `launch.json` to the repository, this part of the readme (create json) is not necessary.

**Review by diarmaidlindsay — CHANGES_REQUESTED** [review]

File: README.md (hunk 9 above)
Comment (dinkar1708) [reply README.md]: nice idea fixed.

**Review by diarmaidlindsay — APPROVED** [review]

**Comment by dinkar1708** [comment]
@enricmacias san can you review please?

**Review by enricmacias — APPROVED** [review]
LGTM
